src/PKSim.Core/Snapshots/Mappers/PopulationMapper.cs

```diff
@@ -1,89 +1,85 @@
-         snapshot.GestationalAge = snapshotRangeFor(randomPopulation.Settings, CoreConstants.Parameter.GESTATIONAL_AGE);
-         snapshot.BMI = snapshotRangeFor(randomPopulation.Settings, CoreConstants.Parameter.BMI);
-         snapshot.AdvancedParameters = snapshotAdvancedParametersFrom(randomPopulation);
+         snapshot.Settings = await _randomPopulationSettingsMapper.MapToSnapshot(randomPopulation.Settings);
+         snapshot.AdvancedParameters = await snapshotAdvancedParametersFrom(randomPopulation);
       }
 
-      private List<AdvancedParameter> snapshotAdvancedParametersFrom(ModelPopulation population)
+      private Task<AdvancedParameter[]> snapshotAdvancedParametersFrom(ModelPopulation population)
       {
-         return population.AdvancedParameters.Select(_advancedParameterMapper.MapToSnapshot).ToList();
+         var tasks = population.AdvancedParameters.Select(_advancedParameterMapper.MapToSnapshot);
+         return Task.WhenAll(tasks);
       }
 
-      private ParameterRange snapshotRangeFor(RandomPopulationSettings randomPopulationSettings, string parameterName) => _parameterRangeMapper.MapToSnapshot(randomPopulationSettings.ParameterRange(parameterName));
-
-      private double? proportionOfFemalesFrom(RandomPopulationSettings randomPopulationSettings)
+      public override async Task<ModelPopulation> MapToModel(SnapshotPopulation snapshot)
       {
-         var female = _genderRepository.Female;
-         return randomPopulationSettings.GenderRatio(female)?.Ratio;
-      }
+         var randomPopulationSettings = await _randomPopulationSettingsMapper.MapToModel(snapshot.Settings);
 
-      private void mapIndividualToSnapshot(SnapshotPopulation snapshot, ModelPopulation population)
-      {
-         snapshot.Individual = _individualMapper.MapToSnapshot(population.FirstIndividual);
+         var population = await _randomPopulationFactory.CreateFor(randomPopulationSettings, CancellationToken.None, snapshot.Seed);
+         MapSnapshotPropertiesToModel(snapshot, population);
+         await updateAdvancedParameters(population, snapshot.AdvancedParameters);
+         return population;
       }
 
-      public override ModelPopulation MapToModel(SnapshotPopulation snapshot)
+      private async Task updateAdvancedParameters(RandomPopulation population, AdvancedParameter[] snapshotAdvancedParameters)
       {
-         throw new NotImplementedException();
+         if (snapshotAdvancedParameters == null)
+            return;
+
+         population.RemoveAllAdvancedParameters();
+         var parameterCache = _parameterTask.PathCacheFor(population.AllIndividualParameters());
+         var tasks = snapshotAdvancedParameters.Select(x => _advancedParameterMapper.MapToModel(x, parameterCache));
+         var advancedParameters = await Task.WhenAll(tasks);
+
+         advancedParameters.Each(x => population.AddAdvancedParameter(x, generateRandomValues: true));
```

Comment:
@Yuri05 Right now there is no way to ensure that the same randomValues are used for advanced parameters (it all depends on the order with which they are added to the collection etc..)
if we really want to also have the same values for those parameters, we need to actually save the double values.
Thoughts?

Reply:
Think it's OK as is.

Reply:
ok cool